src/System.ComponentModel.Annotations/tests/MaxLengthAttributeTests.cs

```diff
@@ -58,6 +86,16 @@ public static void Ctor_Int(int length)
         }
 
         [Theory]
+        [MemberData(nameof(ValidValues_ICollection))]
+        [MemberData(nameof(InvalidValues_ICollection))]
+        [SkipOnTargetFramework(~TargetFrameworkMonikers.NetFramework, "MaxLengthAttribute in the full .NET framework doesn't support ICollection.Count. See https://github.com/dotnet/corefx/pull/2650")]
+        public void Validate_ICollection_NetFx_ThrowsInvalidCastException(MaxLengthAttribute attribute, object value)
```

Comment:
Sorry I'm not sure I understand (this also applies to the comment about the confusion above)

It/when netfx adds the ICollection feature, then this test that is skipped on non-netfx frameworks (I.e runs for netfx) will fail.

The respective netcoreapp test uses the Invalid/Valid_ICollection_TestData in Valid/InvalidValues()

Maybe I could clarify this with a comment?